cloudinit/net/network_state.py

```diff
@@ -806,6 +806,10 @@ def _add_dhcp_overrides(overrides, subnet):
 
         # v2 routes are bound to the interface, in v1 we add them under
         # the first subnet since there isn't an equivalent interface level.
+        # Ordering of subnet config for ipv4 vs ipv6 definitions does not
+        # matter on internal net state. The network renderers, such as netplan,
+        # emit that network config up in the specific interface scope instead
+        # of individual subnet addresses.
```

Comment:
This confuses me.  This comment is specifically about routes:

V2 looks like
```
ethernets:
    ens3:
        dhcp4: True
         addresses: [192.168.41.3/24]
          routes:
            - to: 192.168.43.0/24
              via: 192.168.0.3
            - to: "::/0"
              via: "2001:cafe:face::1/128"
              scope: link
```
and equivalent v1 looks like

```
config:
   - type: physical
     name: ens3
     subnets:
        - type: dhcp4
          routes: 
            - gateway: 192.168.0.3
               network: 192.168.43.0/24
            - gateway: 2001:cafe:face::1/128
               network: ::
       - type: static
          address: 192.168.41.3/24
```

As you can see the v2 routes are put under the *first* subnet for the current interface.
And it's always been the case that the ordering of routes (v4 or v6 ,or both) does not have an impact
on rendering;  hence my confusion.

Reply:
ahh I misread the unit test and completely misinterpreted the output. dropping that useless comment.